src/main/java/org/myrobotlab/framework/MethodCache.java

```diff
@@ -553,7 +547,7 @@ public Object[] getDecodedJsonParameters(Class<?> clazz, String methodName, Obje
       Class<?>[] paramTypes = possible.get(p).getParameterTypes();
       try {
         for (int i = 0; i < encodedParams.length; ++i) {
-          if (encodedParams[i].getClass() == LinkedTreeMap.class) {
+          if (CodecUtils.JSON_DEFAULT_OBJECT_TYPE.equals(encodedParams[i].getClass())) {

```

Comment:
is it a LinkedTreeMap ?

Reply:
Gson uses a LinkedTreeMap while Jackson uses some other tree map, this code here ensures we don't rely on a specific implementation's default type